charts/akeyless-api-gateway/templates/deployment.yaml

```diff
@@ -13,8 +19,7 @@ metadata:
   {{- toYaml .Values.deployment.annotations | nindent 4 }}
   {{- end }}
 spec:
-
-  {{- if not .Values.HPA.enabled }}
+  {{- if and (eq .Values.kind "Deployment") (not .Values.HPA.enabled) }}
```

Comment:
what with HPA yaml?
If set to DaemonSet the HPA should not be applied on YAML level as well

Reply:
You are right i'll fix it